packages/docusaurus-preset-shiki-twoslash/docusaurus-theme-shiki-twoslash/theme/CodeBlock/styles.css

```diff
@@ -0,0 +1,290 @@
+    repeat-x bottom left;
+  padding-bottom: 3px;
+}
+
+pre .query {
+  margin-bottom: 10px;
+  color: var(--ifm-color-primary);
+  display: inline-block;
+}
+
+/** 
+  * In order to have the 'popped out' style design and to not break the layout
+  * we need to place a fake and un-selectable copy of the error which _isn't_ broken out
+  * behind the actual error message.
+  * This sections keeps both of those two in in sync  
+  */
+
+pre .error,
+pre .error-behind {
+  margin-top: 8px;
+  padding: 6px;
+  padding-left: 14px;
+  margin-left: calc(var(--ifm-pre-padding) * -1 - 2px);
+  width: calc(100% + 2px);
+  white-space: pre-wrap;
+  display: block;
+}
+
+pre .error {
+  position: absolute;
+  background-color: #fee;
+  border-left: 2px solid var(--ifm-color-danger-dark);
+  /* Give the space to the error code */
+  display: flex;
+  align-items: center;
+  color: black;
+}
+
+pre .error .code {
+  display: none;
+}
+
+pre .error-behind {
+  user-select: none;
+  color: #fee;
+}
+
+/* Queries */
+pre .arrow {
+  /* Transparent background */
+  background-color: var(--ifm-color-emphasis-200);
+  position: relative;
+  top: -7px;
+  margin-left: 0.1rem;
+  /* Edges */
+  border-left: 1px solid var(--ifm-color-emphasis-200);
+  border-top: 1px solid var(--ifm-color-emphasis-200);
+  transform: translateY(25%) rotate(45deg);
+  /* Size */
+  height: 8px;
+  width: 8px;
+}
+
+pre .popover {
+  margin-bottom: 10px;
+  background-color: var(--ifm-color-emphasis-200);
+  display: inline-block;
+  padding: 0 0.5rem 0.3rem;
+  margin-top: 10px;
+  border-radius: 3px;
+}
+
+/* Completion */
+pre .inline-completions ul.dropdown {
+  display: inline-block;
+  position: absolute;
+  width: 240px;
+  background-color: var(--ifm-color-emphasis-200);
+  padding-top: 5px;
+  font-family: var(--code-font);
+  margin: 0 0 0 3px;
+  padding: 0;
+  border-left: 2px solid var(--ifm-color-primary);
+}
+
+pre .inline-completions ul.dropdown::before {
+  background-color: var(--ifm-color-primary);
+  width: 2px;
+  position: absolute;
+  top: -1.2rem;
+  left: -2px;
+  content: " ";
+}
+
+pre .inline-completions ul.dropdown li {
+  overflow-x: hidden;
+  padding-left: 4px;
+  margin-bottom: 4px;
+}
+
+pre .inline-completions ul.dropdown li.deprecated {
+  text-decoration: line-through;
+}
+
+pre .inline-completions ul.dropdown li span.result-found {
+  color: var(--ifm-color-primary);
+}
+
+pre .inline-completions ul.dropdown li span.result {
+  width: 100px;
+  color: black;
+  display: inline-block;
+}
+
+data-lsp {
+  /* Ensures there's no 1px jump when the hover happens */
+  border-bottom: 1px dotted transparent;
+  /* Fades in unobtrusively */
+  transition-timing-function: ease;
+  transition: border-color 0.3s;
+}
+
+/* Respect people's wishes to not have animations */
+@media (prefers-reduced-motion: reduce) {
+  data-lsp,
+  pre .code-container > a,
+  pre.shiki div.dim,
+  pre.shiki div.highlight::before {
+    transition: none;
+  }
+}
```

Comment:
This is the only thing which I'd like to see improvements on, it's not blocking though.

Can we automate copying the twoslash CSS [from this file](https://github.com/shikijs/twoslash/blob/main/packages/shiki-twoslash/style.css) (`/packages/shiki-twoslash/style.css`) into the bottom of this file during a deploy? I'll _definitely_ forget to update this CSS as things change.

Reply:
I don't think it's gonna work though. My commit just a min ago even modify the default CSS soo much it's basically not the same. Things had to be done to make it work with docusaurus's default infima CSS variables and paddings.

Reply:
OK, then we'll have to think of this as potential tech debt in that we accept that they could get out of sync 👍🏻